### PR TITLE
Fix service name resolving (service name clash)

### PR DESCRIPTION
**Pull request by netoneko**

### CODE REVIEW

File: strelets/adapter/swarm_prepare_reverse_proxy.go [hunk 1]
```diff
@@ -13,7 +13,8 @@ type ReverseProxyConfig struct {
 }
 
 func (d *dockerSwarm) PrepareReverseProxy(ctx context.Context, config *ReverseProxyConfig) (Runner, error) {
-	if err := d.RemoveContainer(ctx, PROXY_CONTAINER_NAME); err != nil {
+	serviceName := GetServiceId(PROXY_CONTAINER_NAME)
```
Comment (electricmonk) [inline strelets/adapter/swarm_prepare_reverse_proxy.go]: @netoneko what was the bug here, and where's the test that reproduces it?

Comment (netoneko) [reply strelets/adapter/swarm_prepare_reverse_proxy.go]: The bug was that the service name passed was not suffixed with `-stack`, therefore when docker looked up `chain-1000` to disable it, it also found `chain-1000001` and it was disabled too.

**Comment by netoneko** [comment]
@electricmonk added the test

**Comment by electricmonk** [comment]
@netoneko if I understand correctly, the issue was that two containers were removed instead of one? Removing is by prefix? Why? It’s something we experienced before, is this the same issue? 

Either way, the test doesn’t reflect the issue...

**Comment by netoneko** [comment]
It’s an underlying issue with how docker does search by service name.

Test shows that the correct service name is provided, one that can’t clash with anything else.

It’s the same issue we have experienced which was fixed with the introduction of GetServiceId method, which wasn’t called in one specific case.

If you have some other test in mind, please provide specific instructions.

**Comment by electricmonk** [comment]
As this is a Docker issue, I think the test needs to be an integration test that includes a Docker environment with two containers having the same prefix

**Comment by netoneko** [comment]
This test would be testing something that we already know.

We know that docker returns two containers that start with the same prefix. The way around that was already implemented - always use full name that can’t clash.

The problem here was that the full name was not passed correctly in a single place. The test covers this issue.

File: test/e2e/e2e_boyar_test.go [hunk 2]
```diff
@@ -131,3 +131,31 @@ func TestE2EAddNewVirtualChainWithSwarmAndBoyar(t *testing.T) {
...
+			cfg.SetKeyConfigPath(fmt.Sprintf("%s/node%d/keys.json", getConfigPath(), i))
+			require.NoError(t, err)
+
+			logger := helpers.DefaultTestLogger()
+			cache := config.NewCache()
+			err = boyar.Flow(context.Background(), cfg, cache, logger)
+			require.NoError(t, err)
+		}
+
+		helpers.WaitForBlock(t, helpers.GetMetricsForPort(9081), 3, WAIT_FOR_BLOCK_TIMEOUT)
```
Comment (electricmonk) [inline test/e2e/e2e_boyar_test.go]: what's the meaning of the port numbers?

File: test/e2e/e2e_boyar_test.go [hunk 3]
```diff
@@ -131,3 +131,31 @@ func TestE2EAddNewVirtualChainWithSwarmAndBoyar(t *testing.T) {
 		helpers.WaitForBlock(t, helpers.GetMetricsForPort(8175), 0, WAIT_FOR_BLOCK_TIMEOUT)
 	})
 }
+
+// Tests boyar.Flow as close as it gets to production starting up
+func TestE2EWithFullFlowAndDisabledSimilarVchainId(t *testing.T) {
+	withCleanContext(t, func(t *testing.T) {
+		for i := 1; i <= 3; i++ {
```
Comment (electricmonk) [inline test/e2e/e2e_boyar_test.go]: what's the purpose of this iteration?

File: test/e2e/e2e_boyar_test.go [hunk 4]
```diff
@@ -131,3 +131,31 @@ func TestE2EAddNewVirtualChainWithSwarmAndBoyar(t *testing.T) {
...
+			logger := helpers.DefaultTestLogger()
+			cache := config.NewCache()
+			err = boyar.Flow(context.Background(), cfg, cache, logger)
+			require.NoError(t, err)
+		}
+
+		helpers.WaitForBlock(t, helpers.GetMetricsForPort(9081), 3, WAIT_FOR_BLOCK_TIMEOUT)
+		helpers.WaitForBlock(t, helpers.GetMetricsForPort(8175), 0, WAIT_FOR_BLOCK_TIMEOUT)
+
+		helpers.Eventually(WAIT_FOR_BLOCK_TIMEOUT, func() bool {
```
Comment (electricmonk) [inline test/e2e/e2e_boyar_test.go]: after employing some detective work, I understand that this checks that chain 100 is offline? if so, can you please make this assertion clearer?

**Pull request merged**